### DISCUSSION TITLE
Yubi keys no longer available for 2FA? Why?

### DISCUSSION THREAD

**Discussion by pa-aros** [community/community#23623] [New to GitHub]
<p>There is a github blog from August 2021, which promotes using Yubi Keys for 2FA.</p><aside class="onebox allowlistedgeneric" data-onebox-src="https://github.blog/2021-08-16-securing-your-github-account-two-factor-authentication/">
  <header class="source">
      <img src="https://github.blog/wp-content/uploads/2019/01/cropped-github-favicon-512.png?fit=32%2C32" class="site-icon" width="32" height="32">

      <a href="https://github.blog/2021-08-16-securing-your-github-account-two-factor-authentication/" target="_blank" rel="noopener nofollow ugc" title="06:00PM - 16 August 2021">The GitHub Blog – 16 Aug 21</a>
  </header>

  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/362;"><img src="https://github.blog/wp-content/uploads/2021/08/GitHub-YubiKey_social-card.jpeg?fit=1200%2C630" class="thumbnail" width="690" height="362"></div>

<h3><a href="https://github.blog/2021-08-16-securing-your-github-account-two-factor-authentication/" target="_blank" rel="noopener nofollow ugc">Securing your GitHub account with two-factor authentication | The GitHub Blog</a></h3>

  <p>The benefits of multifactor authentication are widely documented, and there are a number of options for using 2FA on GitHub.</p>

  <p>
    <span class="label1">Est. reading time: 3 minutes</span>
  </p>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>Just tried to enable 2FA on my private github account and it is not offered any more.</p>
<p>I checked on my company account, where we activated YubiKeys for a group of developers in summer this year and I can see, that the option “security key” is activated there.</p>
<p>We like to have this option for a new developer and now YubiKeys are no longer offered. Our company policy does not allow the usage of personal equipment (smartphone) so Yubi Key was a good and cheap choice.</p>
<p>Can anybody explain why this option is no longer available and if it will become available again in the near future?</p>
<p>Thank you</p>

**Comment by joshuawalker** [comment, marked as answer]
<p>Hey <a class="mention" href="/anrose00">@anrose00</a>—you can absolutely still add a YubiKey, or other security key device to your account but you’ll need to add an authenticator app or SMS number as your primary method, first:</p>
<aside class="onebox allowlistedgeneric" data-onebox-src="https://docs.github.com/en/authentication/securing-your-account-with-two-factor-authentication-2fa/configuring-two-factor-authentication">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">

      <a href="https://docs.github.com/en/authentication/securing-your-account-with-two-factor-authentication-2fa/configuring-two-factor-authentication" target="_blank" rel="noopener">GitHub Docs</a>
  </header>

  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/3141202/181078297-419cc58b-4759-4bab-a1b0-367266341a41.png" class="thumbnail onebox-avatar" width="500" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_500x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_750x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_1000x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_10x10.png">

<h3><a href="https://docs.github.com/en/authentication/securing-your-account-with-two-factor-authentication-2fa/configuring-two-factor-authentication" target="_blank" rel="noopener">Configuring two-factor authentication - GitHub Docs</a></h3>



  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

**Comment by pa-aros** [comment]
<p>Thanks for your response. But initially this restriction did not exist. The question is why?<br>
Accounts that were created at that time, don’t have any other 2FA methods and there is no problem with that.</p>
<p>Any explanation?</p>

**Comment by joshuawalker** [comment]
<p>It’s always been the case that to add a security key, you need an authenticator app or SMS number configured—we haven’t changed anything there. This was true way back when we added support for FIDO U2F in 2015, and now with our support for WebAuthn:</p>
<aside class="onebox allowlistedgeneric" data-onebox-src="https://github.blog/2019-08-21-github-supports-webauthn-for-security-keys/#the-future-of-authentication-secure-and-easy-to-use">
  <header class="source">
      <img src="https://github.blog/wp-content/uploads/2019/01/cropped-github-favicon-512.png?fit=32%2C32" class="site-icon" width="32" height="32">

      <a href="https://github.blog/2019-08-21-github-supports-webauthn-for-security-keys/#the-future-of-authentication-secure-and-easy-to-use" target="_blank" rel="noopener" title="06:11PM - 21 August 2019">The GitHub Blog – 21 Aug 19</a>
  </header>

  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/362;"><img src="https://user-images.githubusercontent.com/3141202/181078300-94f1e01b-72b3-4368-82ca-5b73aa898db3.jpeg" class="thumbnail" width="690" height="362" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/6/f/6f62d5dc0dbed49465f7b63fc695ad3a1e4f8d0d_2_690x362.jpeg, https://aws1.discourse-cdn.com/github/optimized/3X/6/f/6f62d5dc0dbed49465f7b63fc695ad3a1e4f8d0d_2_1035x543.jpeg 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/6/f/6f62d5dc0dbed49465f7b63fc695ad3a1e4f8d0d_2_1380x724.jpeg 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/6/f/6f62d5dc0dbed49465f7b63fc695ad3a1e4f8d0d_2_10x10.png"></div>

<h3><a href="https://github.blog/2019-08-21-github-supports-webauthn-for-security-keys/#the-future-of-authentication-secure-and-easy-to-use" target="_blank" rel="noopener">GitHub supports Web Authentication (WebAuthn) for security keys</a></h3>

  <p>The WebAuthn standard for security keys is making authentication as easy as possible. Now you can use security keys for second-factor authentication on GitHub with many more browsers and devices.</p>

  <p>
    <span class="label1">Est. reading time: 2 minutes</span>
  </p>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>Removing this requirement and making security keys a primary method of authentication is certainly something we’re thinking about but, for now, keep your app or SMS number nearby.</p>

**Comment by msct** [comment]
<blockquote>
<p>but you’ll need to add an authenticator app or SMS number as your primary method, first</p>
</blockquote>
<p>A problem is that GitHub’s documentation often doesn’t mention this.  The interface <em>is</em> confusing, and the policy is confusing too.</p>
<blockquote>
<p>Removing this requirement and making security keys a primary method of authentication is certainly something we’re thinking about</p>
</blockquote>
<p>Context here is in regards to second-factor authentication.</p>
<p>For instance, I currently don’t have (or want) a phone line.  WiFi is everywhere I go. The Authenticator apps I’m familiar with require an SMS channel.  It’d sure be nice if GitHub <em>directly</em> supported second-factor security with WebAuthn devices (like Yubikeys, embedded system chips, etc) without a traditional telephone line.</p>
<p>There isn’t a foundational technical problem preventing GitHub’s support. And since GitHub generates one-time-use recovery codes, there isn’t really user lockout experience problem with it either.  --I have my Google (banks, hosting, etc) accounts secured in this exact manor; their interfaces tend to <em>recommend</em> multiple hardware keys, in addition to recovery codes, but it <em>is</em> unusual in the industry to gate-keep webauthn.</p>
<p>Cheers.</p>

**Comment by airtower-luna** [comment]
<aside class="quote no-group" data-username="msct" data-post="5" data-topic="210999">
<div class="title">
<div class="quote-controls"></div>
 msct:</div>
<blockquote>
<p>The Authenticator apps I’m familiar with require an SMS channel.</p>
</blockquote>
</aside>
<p>TOTP 2FA does not, it doesn’t require any network connection at all after initial setup. But I agree that direct WebAuthn would be nice. :slightly_smiling_face:</p>

**Comment by mlane3** [comment]
<p>That <strong>policy</strong> will need to be changed slightly probably long term.  As the Salesforce roll out of there MFA shows newer 2FA and MFA requirements breakdown for service accounts and automation accounts.</p>
<p>In theory Github Actions, Model Ops --anything with Continious Integration <strong>should break</strong> if its using SMS to connect.   The reason I point to the Salesforce roll out, is that it a example of there One Time Password 2FA kind of failed due to confusion of the user base.</p>
<p>The lesson learned was that there are not factors options for users for authentication with 2FA.  Not every authenticator app was supported.</p>
<p>What should be changed – more options for 2FA with Yubi keys not just mobile app, but location.  The increase in the number of available authenticator apps and vendors especially those that allow github actions.</p>
